lib/Sema/TypeCheckNameLookup.cpp

```diff
@@ -881,15 +882,29 @@ class MissingImportFixItCache {
 
     // Add an appropriate access level as long as it would not conflict with
     // existing imports that lack access levels.
-    if (!foundImport || anyImportHasAccessLevel)
-      info.accessLevel = sf.getMaxAccessLevelUsingImport(mod);
+    auto accessLevelForImport = [&]() -> std::optional<AccessLevel> {
+      auto maxAccessLevel = sf.getMaxAccessLevelUsingImport(mod);
+      if (internalImportsByDefaultEnabled) {
+        if (!foundImport && maxAccessLevel <= AccessLevel::Internal)
```

Comment:
Do we need to check for imports in other files here? In internal-imports-by-default mode the access-level in other files shouldn't mater, the inconsistency diagnostics is disabled in that mode.

Reply:
My thinking is that we should follow the style that the developer is already using. If they write explicit access levels on imports of the module in other files, then the fixit should too (which is handled by the logic below).

Reply:
Ah ok, that makes sense.